py4DSTEM/process/diskdetection/diskdetection.py

```diff
@@ -70,6 +71,12 @@ def find_Bragg_disks_single_DP_FK(DP, probe_kernel_FT,
                                             'multicorr': uses the multicorr algorithm with
                                                         DFT upsampling
         upsample_factor      (int) upsampling factor for subpixel fitting (only used when subpixel='multicorr')
+        filter_function      (callable: lambda or partial function of one argument) filtering function to apply
+                             to each diffraction pattern before peakfinding. must be a function of only
+                             one argument (the diffraction pattern) and able to be pickled by the dill library.
+                             This is useful for doing noise reduction or binning on the fly for memory mapped
+                             datasets or for parallel operation. The shape of the returned DP must match the shape
+                             of the probe kernel.
```

Comment:
A few things!
- Is there a reason we're not just calling this a function?  I'm not sure what is meant by "partial function of one argument" here - do you mean that it only accepts one argument?  If so this is specified later in the docstring.  It also seems to me like it shouldn't really matter if it's a lambda function or not.  I think the language here might be unnecessarily confusing. 
- pickling is only relevant for distribution to parallel processing, yes?  Assuming this is true,this should also be specified, at the end of the docstring.  Same design principle as above: best to keep things simple wherever possible, start with simple cases, and advance to more complex ones.  I suspect this would trip up plenty of users who are working locally and don't know what this means.
- "This is useful for..." I would leave this exposition out of the docstring.  We can show this in examples.

E.g. I'd suggest:
```
(function) filtering function to apply
 to each diffraction pattern before peakfinding. must be a function of only
 one argument (the diffraction pattern), which returns a diffraction pattern.
The shape of the returned DP must match the shape of the probe kernel
(important e.g. if the filter function is used for binning).  If used with parallelized
disk detection, the function must be able to be pickled by the dill library.
```